test/integration/testManualConnection.js

```diff
@@ -70,6 +72,40 @@ if (process.env.RUN_MANUAL_TESTS_ONLY === 'true') {
           }
         });
       });
+
+      describe('Connection - ID Token authenticator', async function (done) {
+        const key = Util.buildCredentialCacheKey(connectionOption.host, connectionOption.username, 'ID_TOKEN');
+        GlobalConfig.getCredentialManager().remove(key);
+        
+        const connectionOption = connOption.externalBrowser;
+
+        it('test - obtain the id token from the server and save it on the local storage', async function () {
+          const connection = snowflake.createConnection(connectionOption);
+          await connection.connectAsync(function (err) {
+            assert.ok(!err);
+            const idToken = GlobalConfig.getCredentialManager().read(key);
+            assert.ok( idToken !== null);
```

Comment:
Could we verify that file contains expected key and some token?

Reply:
This testing is the manual testing. The purpose of this testing is that we need to verify that the ID token connection is working with its own credentials. As a result, we cannot know the expected key and tokens in this testing. I created the unit testing for the JSON credential manager in a different file.